tests/recipes/recipe_lib_test.py

```diff
@@ -0,0 +1,151 @@
+from unittest import mock
+from tests.recipes.recipe_ctx import RecipeCtx
+
+
+class BaseTestForMakeRecipe(RecipeCtx):
+    """
+    An unittest for testing any recipe using the standard build commands
+    (`configure/make`).
+
+    .. note:: Note that Some cmake recipe may need some more specific testing
+        ...but this should cover the basics.
+    """
+
+    recipe_name = None
+    recipes = ["python3", "kivy"]
+    recipe_build_order = ["hostpython3", "python3", "sdl2", "kivy"]
+    expected_compiler = (
+        "{android_ndk}/toolchains/llvm/prebuilt/linux-x86_64/bin/clang"
+    )
+
+    sh_command_calls = ["./configure"]
+    """The expected commands that the recipe runs via `sh.command`."""
+
+    extra_env_flags = {}
+    """
+    This must be a dictionary containing pairs of key (env var) and value.
+    """
+
+    def __new__(cls, *args):
+        obj = super(BaseTestForMakeRecipe, cls).__new__(cls)
+        if obj.recipe_name is not None:
+            print(f"We are testing recipe: {obj.recipe_name}")
+            obj.recipes.append(obj.recipe_name)
+            obj.recipe_build_order.insert(1, obj.recipe_name)
+            return obj
```

Comment:
I guess you're using the `__new__` over the `__init__()` for a reason, but I have to ask. Why can't this be dealt with  it? Something like:
```python
def __init__(self, *args, **kwargs):
    if self.recipe_name is not None:
...
```

Reply:
Aaa... I faced several issues along the way with this: duplicated tests, tests not initialized...:laughing:..finally using `__new__` is what it worked and I forgot about this lines until now...

I just made a test with `__init__`  and it also can work so I'm considering to change `__new__` with `__init__` (I think it's better to not mess with `__new__` if we can do it with `__init__`)

...so...@AndreMiras, let's go with `__init__`?

Reply:
> I think it's better to not mess with `__new__` if we can do it with `__init__`

That recaps my feeling pretty well. I'm not against `__new__` in general, but it needs to be clearly justified. But yes again if you can't make it work consistently with `__init__` then I'm go for it.